spark-cassandra-connector/src/main/scala/com/datastax/spark/connector/util/ConfigCheck.scala

```diff
@@ -29,7 +32,11 @@ object ConfigCheck {
    * @param conf SparkConf object to check
    */
   def checkConfig(conf: SparkConf): Unit = {
-    val unknownProps = getUnknownProperties(conf)
+    val connectionFactory = CassandraConnectionFactory.fromSparkConf(conf)
```

Comment:
These should be an apply but i know you disagree.
Apply's are for creating, the signature of the applys is all one needs, not "fromSparkConf"